Properly link quazip_static against its dependencies

From the looks of r256, it appears that the target_link_libraries() call in
quazip/CMakeLists.txt was modified with the intention of linking both
${QUAZIP_LIB_TARGET_NAME} and quazip_static against all required libraries.
However, target_link_libraries()'s syntax is such that the first argument is
the library that all other arguments will be linked against. In other words,
libquazip(5).so will link against quazip_static, which is not what is desired.

Creator: Raphael Kubo da Costa

https://sourceforge.net/p/quazip/patches/28/

Index: quazip/CMakeLists.txt
===================================================================
--- quazip/CMakeLists.txt.orig	2016-01-03 06:01:40 UTC
+++ quazip/CMakeLists.txt
@@ -25,7 +25,8 @@ endif ()
 
 set_target_properties(${QUAZIP_LIB_TARGET_NAME} quazip_static PROPERTIES VERSION 1.0.0 SOVERSION 1 DEBUG_POSTFIX d)
 # Link against ZLIB_LIBRARIES if needed (on Windows this variable is empty)
-target_link_libraries(${QUAZIP_LIB_TARGET_NAME} quazip_static ${QT_QTMAIN_LIBRARY} ${QTCORE_LIBRARIES} ${ZLIB_LIBRARIES})
+target_link_libraries(quazip_static ${QT_QTMAIN_LIBRARY} ${QTCORE_LIBRARIES} ${ZLIB_LIBRARIES})
+target_link_libraries(${QUAZIP_LIB_TARGET_NAME} ${QT_QTMAIN_LIBRARY} ${QTCORE_LIBRARIES} ${ZLIB_LIBRARIES})
 
 install(FILES ${PUBLIC_HEADERS} DESTINATION include/quazip${QUAZIP_LIB_VERSION_SUFFIX})
 install(TARGETS ${QUAZIP_LIB_TARGET_NAME} quazip_static LIBRARY DESTINATION ${LIB_DESTINATION} ARCHIVE DESTINATION ${LIB_DESTINATION} RUNTIME DESTINATION ${LIB_DESTINATION})
